### ISSUE TITLE
Show item in chat

### ISSUE THREAD

**Issue by Sikatsu** [kangarko/ChatControl-Pro#1054]
Hey, would it be possible to add a feature so you can show the item you are holding in the chat?
For example when you write [Item] it will state the name and you can hover over it for more information.

**Comment by Sikatsu** [comment]
I will try that, however i'm not familiar with it. Does that mean they need to write {player_hand} in the chat?

**Comment by kangarko** [comment]
Not exactly so, but there are two ways of getting a somewhat 

1) You can create a chat variable that will transform anything, in the case below {player_hand}, into that variable. However, JSON is not possible:

![a](https://i.imgur.com/tRzls6f.png) 

2) Or you can create own JSON and inside of that json use {player_hand} in the hover event. Possible, but more complicated, and also requires sending a completely new message.

https://github.com/kangarko/ChatControl-Pro/wiki/JSON#json-in-rules

I understand there are complicated measurements, however due to the nature of in-game chat since MC 1.7.10 it is not easily possible to interrupt the JSON message the chat consists of.

**Comment by Sikatsu** [comment]
Not sure if I kind of understand on what to do. I think it kind of needs JSON so you can hover and see the stats? Like see the enchantments or names/lores of items?

**Comment by kangarko** [comment]
Yes, you can use this webpage to generate a fancy message: https://minecraftjson.com/

For example, you can use this rule to send it:

```
match \{player_hand\}
then deny
then console tellraw {player} ["",{"text":"Hover Me for Item Name","hoverEvent":{"action":"show_text","value":{"text":"","extra":[{"text":"{player_hand}"}]}}}]
```

**Comment by Sikatsu** [comment]
Hey, I still don't exactly know how to implement this. I added those lines to variables/javascript.txt, but what do I have to do in formatting.yml? And what are the next steps exactly?

**Comment by kangarko** [comment]
Hey, so here is a complete tutorial:

**1. Install your new variable in variables/javascript.txt**
```yaml
define {player_hand}
script:
  
  // Get the item
  var item = player.getItemInHand();
  
  // Return readable version (example: DIAMOND_SWORD is returned as Diamond Sword)
  prettify(item);
    
  function prettify(item) {
    var utils = Java.type("org.apache.commons.lang.WordUtils");  
    var itemName = item.getType().toString();
    var fancyName = utils.capitalizeFully( itemName.toLowerCase().replaceAll("_", " ") )
    
    return fancyName == "Air" ? "None" : fancyName;
  }
```

**2. Use the {player_hand} variable normally in formatting.yml**
![a](https://i.imgur.com/qpFdnNs.png)

**3. Enjoy it in the game!**
![a](https://i.imgur.com/s2CJmt0.png)

**Comment by Sikatsu** [comment]
Thanks, however that is not quite as what i'm trying to achieve.

What I mean is something like this:
https://www.spigotmc.org/resources/chatitem-display-items-in-chat.19064/

Where if you typ [item] in the chat it will then show a chat message and players can hover over it and see all enchantments, lores ect.

**Comment by kangarko** [comment]
I know, but as we discussed above the chat has since MC 1.7.10 been split into parts that are not so easy to decode, edit and build back again. That's why there is no basic support for that. You can take a look at how that plugin you mentioned must manually decode the json: https://github.com/dadus33/ChatItem/blob/master/src/main/java/me/dadus33/chatitem/json/JSONManipulatorCurrent.java I currently don't posses this experience nor time required for this solution, sorry.
